### DISCUSSION TITLE
nparticles vs nparticle_per_cell

### DISCUSSION THREAD

**Discussion by erny123** [epochpic/epoch#296] [Q&A] [answered]
Hi EPOCH developers. 

I am wondering what the difference is between `nparticles` and `nparticles_per_cell`. 

I am running some over dense simulations where the debye length is less than 1e-11 for 300K temperature. 
Using` nparticles_per_cell = 1 `forces me to use over 1.5 million particles which is just too much for a quick survey for results. 

However, simulating only a single species of electrons (no background species or immobile species) if I use: 
`nparticles = 250000`

Then the results between simulation are completely different. 

Here's my thoughts: 
`nparticles_per_cell ` evenly distributes the particles on the grid. The simulation starts off with full neutrality (despite there being only electron species) and evolves properly from there because there is an evenly distributed "positive background" from initial neutrality. 

Using `nparticles` randomly distributes the particles on the grid. This sets up an initial neutrality based of the random positions of the particles and so does not have an evenly distributed "positive background". As the simulation evolves, the randomly distributed "positive background" causes a lack of proper physics to take place. 

I am currently testing this by using `nparticles` electron species and another ion species set to be background with `background_species = T`

EDIT: 

Here's the results from simulation using `nparticles_per_cell` (first graph) vs `nparticles` (second graph). 

![image](https://user-images.githubusercontent.com/34816295/135887607-86a01614-39cc-4aa1-9f90-a5aac4ca1f6a.png)

![image](https://user-images.githubusercontent.com/34816295/135887437-0ce73668-a518-47ed-9164-4c1ab1b6e5a9.png)

With an example input deck: 

```
begin:control
  nx = 417944

  # Size of domain 

  x_min = 0 * micron 

  x_max = 50*micron 

 

  # Final time of simulation 

  t_end = 1*pico 

end:control 

begin:boundaries 
  bc_x_min = simple_laser 

  bc_x_max = simple_outflow 

  bc_x_max_particle = heat_bath

end:boundaries 

begin:laser 

  boundary = x_min 

  intensity_w_cm2 = 1.4e18 * (1/1.05)^2 * (0.1)^2 

  lambda = 1.050 * micron 

  t_profile = gauss(time, 90*femto, 60*femto) 

  t_end = 200*femto 

end:laser 


begin:laser 

  boundary = x_min 

  intensity_w_cm2 = 1.4e18 * (1/(1.530)^2 * (0.1)^2 

  lambda = 1.530 * micron 

  t_profile = gauss(time, 90*femto, 60*femto) 

  t_end = 200*femto 

end:laser 

begin:species 

  name = electron 

  charge = -1.0 

  mass = 1.0 

  temp = 300 

  number_density = 0.1e27 

  #nparticles_per_cell = 1
  nparticles = 417944

end:species 

begin:dist_fn 

 name = Px

 ndims = 1 

 direction1 = dir_px

 range1 = (0,0) 

 resolution1 = 4096 

 include_species: electron 

end:dist_fn 

begin:dist_fn 

 name = Energy 

 ndims = 1 

 direction1 = dir_en 

 range1 = (0,0) 

 resolution1 = 4096 

 include_species: electron 

end:dist_fn 

 
begin:output 

  #name = normal   

  dt_snapshot = 1*femto 

  distribution_functions = always 

  px = always  

  # Properties on grid 

  grid = always 
  temperature = always

  ey = always 

  ex = always 

  number_density = always

end:output 

```

**Comment by erny123** [comment, marked as answer]
After talking to others, it seems to be more of a convergence issue. There just isn't enough particles to reach convergent results.

In the second simulation, setting the number of particles to 2x the original number **(picture 2):**
`nparticles = 835888`

Gives the same results as setting **(picture 1)**
`nparticles_per_cell =1`


![image](https://user-images.githubusercontent.com/34816295/135901429-8151ca20-d926-4929-8ea3-1a0575650f18.png)

![image](https://user-images.githubusercontent.com/34816295/135901444-2e225e58-ae5f-490d-ab72-8954503d81cc.png)

**Comment by TomGoffrey** [comment]
Some comments:

1. The difference between `nparticles` and `nparticles_per_cell` is that if you specify `nparticles` you are specifying the total number of particles, whereas if you are specifying `nparticles_per_cell` you are specifying the number of particles per cell, so the total number of particles in your simulation will be `nparticles_per_cell` * nx (in your case as your simulation is 1D).
2. As you say above, either way, you must check that your simulation is well resolved by the number of particles you are using. If your results are (strongly) dependent on the number of particles you are using then the simulation is not converged and your answer is likely not correct.
3. In either case, EPOCH loads the particles randomly in each cell. It doesn't load them in a uniform lattice.
4. Unless you start with a non-zero electric field you are implicitly assuming your simulation is initially charge neutral. Even if you only include electrons in your simulation what you are effectively simulating is the evolution of those electrons in the presence of an immobile neutralising background. This is discussed in #158 and #221.

**Reply by erny123** [reply]
Got it. Thank you, Tom.
